upup/pkg/fi/cloudup/apply_cluster.go

```diff
@@ -1056,6 +1056,23 @@ func (c *ApplyClusterCmd) addFileAssets(assetBuilder *assets.AssetBuilder) error
 			c.Assets[arch] = append(c.Assets[arch], mirrors.BuildMirroredAsset(u, hash))
 		}
 
+		if c.Cluster.UsesExternalECRCredentialsProvider() {
+			binaryLocation := c.Cluster.Spec.CloudProvider.AWS.BinaryLocation
+			if binaryLocation == nil {
+				binaryLocation = fi.PtrTo("https://artifacts.k8s.io/binaries/cloud-provider-aws/v1.27.1")
```

Comment:
Should `v1.27.1` be something like AWS CCM version? Not sure if we have access to that here.

Reply:
Its a good catch. I was thinking of a helper method here where we feed in the K8s version and returns the cloud provider version. And move the case statement from CCM into that function. But I thought there wasn't a rush to do this right now.

Reply:
No rush.